weave/trace_server_bindings/remote_http_trace_server.py

```diff
@@ -193,6 +222,7 @@ def _generic_request_executor(
             data=req.model_dump_json(by_alias=True).encode("utf-8"),
             auth=self._auth,
             stream=stream,
+            timeout=DEFAULT_TIMEOUT,
```

Comment:
Without this timeout it's technically possible for requests to permanently hang and block the processor.  This can cause the queue to back up and OOM the system